### DISCUSSION TITLE
Issue installing IDAES

### DISCUSSION THREAD

**Discussion by chris-hampel-CA** [IDAES/idaes-pse#613] [Q&A]
Hi everyone, 

I ran into an issue when installing the IDAES software and am hoping to hear if anyone has come across this error. 

When starting at the **Generic Install** section on the instructions [https://idaes-pse.readthedocs.io/en/stable/getting_started/index.html#min-install-generic](url), I ran step 1 in anaconda prompt `pip install idaes-pse`. After this I received an error in step 2 `idaes get-extensions`. The error read back as: 

_from pyutilib.excel.spreadsheet import ExcelSpreadsheet, Interfaces
ImportError: cannot import name 'Interfaces' from 'pyutilib.excel.spreadsheet' (c:\users\christopher hampel\anaconda3\lib\site-packages\pyutilib\excel\spreadsheet.py)_

This error appears when attempting to run a cell that imports pyomo packages in the `introduction_exercise.ipynb` jupyter notebook. 

In addition, interestingly, when doing the other install method (**Install IDAES using CONDA**) the error with _'Interfaces'_ never appears. However, the error still appears when attempting to run the `introduction_exercise.ipynb` jupyter notebook. 

Does anyone have any suggestions to fix this?

**Comment by andrewlee94** [comment]
Hi @chris-hampel-CA ,

I haven't had the time to look into this in detail, but I think the issue is due to `introduction_exercise.ipynb` not having been updated recently to reflect the deprecation of `pyutilib` as a dependency of IDAES and Pyomo. Thank you for bringing this to our attention, as it seems it slipped through our automated testing of the examples.

**Reply by chris-hampel-CA** [reply]
Thanks Andrew.

Just an update. I do not have issues running `introduction_exercise.iypnb` if I switch the kernel to `my-idaes-env` rather than the base environment (this is due to my second install attempt using the successful `conda` method rather than using the unsuccessful `pip install` method.

**Comment by lbianchi-lbl** [comment]
Hi @chris-hampel-CA, thanks again for the report, I'm looking into this. Could you point me to the version of the `introduction_exercise.ipynb` notebook you were using when the error occurred?

**Reply by chris-hampel-CA** [reply]
idaes, version 1.11.0

**Reply by lbianchi-lbl** [reply]
OK, thanks. And what about the `introduction_exercise.ipynb` itself? e.g., how did you run it? Was it from a file in a local folder? If so, how was it downloaded? Did you use the `idaes get-examples` command? I'm trying to get as much information as I can on the actual notebook you tried to run when the error occurred to try and reproduce the issue.

If you prefer, and you have access to the notebook file, you should be able to attach it to a message in this discussion by dragging and dropping it into the GitHub reply text box.

**Reply by chris-hampel-CA** [reply]
Yes, the file came from `idaes get-examples`. I opened the notebook in both VS code and the Jupyter Notebook application. It would not run the cell that imports packages, 

`from pyomo.environ import ConcreteModel, Var
model = ConcreteModel()
model.x = Var()
model.y = Var()`

This does run however if I change the kernel to the `my-idaes-env`. 

The root of the problem seems to be the unsuccessful `idaes get-extensions` command when following the steps in [https://idaes-pse.readthedocs.io/en/stable/getting_started/index.html#min-install-generic](url) related to `pyutilib`

(P.S. .iypnb files are not supported for pasting into chats)

**Reply by andrewlee94** [reply]
Looking a bit deeper, this looks like it might be an issue on your end @chris-hampel-CA. Have you ever installed an earlier version of Pyomo before that might be in your base environment somewhere? The `introduction_exercise.ipynb` notebook itself is not importing `pyutilib`, so I am wondering if you have an older version of Pyomo in the background that is trying to import `pyutilib` instead.

**Reply by lbianchi-lbl** [reply]
I agree with @andrewlee94 that this could be due to the environment rather than the notebook itself, especially after you mention changing the kernel, as sometimes it can be hard to understand what Python executable/environment is used for the default kernel (typically displayed as "Python 3" in the Jupyter kernel selection menus).

One way to test Andrew's hypothesis that the default kernel is using an older version of Pyomo would be to run e.g. `!pip list` in a cell of a notebook running the default kernel, and compare the result after running the same cell with the `my-idaes-env` kernel.

**Reply by chris-hampel-CA** [reply]
I had a different version of Pyomo already installed in my base environment before the IDAES install. It was previously Pyomo 5.6.8. This was uninstalled during the IDAES install. Now, both my base and idaes envirnoments show Pyomo 6.2 as the installed version after using the `!pip list` command. 


My anaconda prompt during the original `pip` installation of IDAES:

_Successfully built bunch
Installing collected packages: pywin32, text-unidecode, pyomo, tinydb, rbfopt, python-slugify, flask-cors, bunch, idaes-pse
  Attempting uninstall: pywin32
    Found existing installation: pywin32 227
    Uninstalling pywin32-227:
      Successfully uninstalled pywin32-227
  Attempting uninstall: pyomo
    Found existing installation: Pyomo 5.6.8
    Uninstalling Pyomo-5.6.8:
      Successfully uninstalled Pyomo-5.6.8
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
switch-model 2.0.6 requires Pyomo<=5.6.8,>=4.4.1, but you have pyomo 6.2 which is incompatible.
Successfully installed bunch-1.0.1 flask-cors-3.0.10 idaes-pse-1.11.0 pyomo-6.2 python-slugify-5.0.2 pywin32-225 rbfopt-4.2.2 text-unidecode-1.3 tinydb-4.5.2_
